lib/httparty/parser.rb

```diff
@@ -114,8 +114,10 @@ def xml
       MultiXml.parse(body)
     end
 
+    UTF8_BOM = "\xEF\xBB\xBF".freeze
+
     def json
-      JSON.parse(body, :quirks_mode => true, :allow_nan => true)
+      JSON.parse(body.gsub(/^#{UTF8_BOM}/, ''), :quirks_mode => true, :allow_nan => true)
```

Comment:
I'm not sure if this is the right spot for this logic, or if I should be checking encoding somehow before trying to strip the BOM.

I'm open to suggestions, if there is a better way to do this.